examples/server.rs

```diff
@@ -0,0 +1,61 @@
+#[cfg(any(
+    all(feature = "noise", feature = "tls"),
+    all(not(feature = "noise"), not(feature = "tls"))
+))]
+fn main() {}
+
+#[cfg(any(
+    all(feature = "noise", not(feature = "tls")),
+    all(not(feature = "noise"), feature = "tls")
+))]
+#[async_std::main]
+async fn main() -> anyhow::Result<()> {
+    use futures::stream::StreamExt;
+    use libp2p_quic::{Keypair, QuicConfig, ToLibp2p};
+
+    #[cfg(feature = "noise")]
+    type Crypto = libp2p_quic::NoiseCrypto;
+    #[cfg(feature = "tls")]
+    type Crypto = libp2p_quic::TlsCrypto;
+
+    tracing_subscriber::fmt()
+        .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
+        .with_max_level(tracing::Level::TRACE)
+        .try_init()
+        .ok();
+    log_panics::init();
+
+    let keypair = Keypair::generate(&mut rand_core::OsRng {});
+    let peer_id = keypair.to_peer_id();
+    let mut listener = QuicConfig::<Crypto>::new(keypair)
+        .listen_on("/ip4/127.0.0.1/udp/8383/quic".parse()?)
+        .await?;
+
+    tracing::info!("{}", peer_id);
+
+    while let Some(Ok(event)) = listener.next().await {
+        use libp2p::core::transport::ListenerEvent::*;
+        match event {
+            NewAddress(addr) => {
+                tracing::debug!("new address: {}", addr);
+            },
+            Upgrade { upgrade, .. } => {
+                let (peer_id, muxer) = upgrade.await?;
+                tracing::debug!("incoming connection: {}", peer_id);
+                use libp2p::core::StreamMuxer;
+                let event = futures::future::poll_fn(|cx| muxer.poll_event(cx)).await;
+                dbg!(&event);
+                // what shall I do with the muxer?
```

Comment:
This one should be the same as in https://github.com/libp2p/go-libp2p-quic-transport/blob/master/cmd/server/main.go#L65, however I am new to libp2p, so it's kinda tricky for me to implement it.
Would you please help me a little bit or it will take some time for me to find out what's going on.

However I got:

```
ConnectionLost {
        reason: ApplicationClosed(
            ApplicationClose {
                error_code: 0,
                reason: Jul 30 15:40:00.645 TRACE async_io::driver: main_loop: sleeping for 50 us    
b"",
            },
        ),
    },
```

when I run it: `go-libp2p-quic-transport/cmd$ go run client/main.go /ip4/127.0.0.1/udp/36591/quic $SERVER_PEER_ID`.

Note: apply this patch to make it work (rust-libp2p identity does not support ECDSA):

```diff
diff --git a/cmd/client/main.go b/cmd/client/main.go
index bc033e8..e7e5060 100644
--- a/cmd/client/main.go
+++ b/cmd/client/main.go
@@ -33,7 +33,7 @@ func run(raddr string, p string) error {
        if err != nil {
                return err
        }
-       priv, _, err := ic.GenerateECDSAKeyPair(rand.Reader)
+       priv, _, err := ic.GenerateEd25519Key(rand.Reader) // GenerateECDSAKeyPair
        if err != nil {
                return err
        }
diff --git a/go.mod b/go.mod
```

Reply:
you shouldn't be messing about with the muxer, did you look at the smoke example? you should pass the transport to the swarm and implement a network behaviour.

Reply:
I took a look at the smoke example. However the original [example](https://github.com/libp2p/go-libp2p-quic-transport/blob/master/cmd/server/main.go) in go does not use swarm, instead it accepts and reads a raw stream of data. I would like to do the same.

Reply:
well, that's not so easy with rust-libp2p, although possible. I guess there aren't any good examples, so you'll have to figure it out.

Reply:
not sure we need the example in libp2p-quic. we could add this as a test if you think it's useful, but we shouldn't have bad examples